src/main/java/cubicchunks/CommonEventHandler.java

```diff
@@ -133,11 +134,11 @@ public void onPlayerJoinWorld(EntityJoinWorldEvent evt) {
     public static class ClientProxy extends ServerProxy {
 
         // shouldSkipWorld
-        @Override public boolean test(ICubicWorld world) {
-            if (super.test(world)) {
-                return true;
-            }
-            if (((World) world).getChunkProvider().getClass() != ChunkProviderClient.class) {
+        @Override
+        public boolean test(ICubicWorld world) {
+            if (!super.test(world))
+                return false;
+            if (world.getClass() == (Class<?>) WorldClient.class && ((World) world).getChunkProvider().getClass() != ChunkProviderClient.class) {
```

Comment:
My thoughts: We load both WorldServer and WorldClient for a single player game ("Client" side).
Therefore there should be ChunkProviderServer as well, isn't it?